src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -112,29 +136,28 @@ public Object instanceMain(final String[] args) {
      * Note: this is the only method that is allowed to call System.exit (because gatk tools may be run from test harness etc)
      */
     protected final void mainEntry(final String[] args) {
+        final CommandLineProgram program = extractCommandLineProgram(args, getPackageList(), getClassList(), getCommandLineName());
         try {
-            final Object result = instanceMain(args);
+            final Object result = runCommandLineProgram(program, args);
             handleResult(result);
             System.exit(0);
         } catch (final CommandLineException e){
-            //the usage has already been printed so don't print it here.
-            if(printStackTraceOnUserExceptions()) {
-                e.printStackTrace();
+            if (program != null) { //program can be null if th
+                System.err.println(program.getUsage());
```

Comment:
Hmn, that's a good point.  I was thinking  that since program could be null and we should guard against that case, but you're right that a null program should never throw a `CommandLineException`.   I'll drop it.